aws/resource_aws_guardduty_publishing_destination_test.go

```diff
@@ -0,0 +1,292 @@
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error retrieving GuardDuty Publish Destinations: %s", err)
+	}
+
+	return sweeperErrs.ErrorOrNil()
+}
+
+func TestAccAwsGuardDutyPublishingDestination_basic(t *testing.T) {
+	resourceName := "aws_guardduty_publishing_destination.test"
+	bucketName := acctest.RandomWithPrefix("tf-acc-test")
+	detectorResourceName := "aws_guardduty_detector.test_gd"
+	bucketResourceName := "aws_s3_bucket.gd_bucket"
+	kmsKeyResourceName := "aws_kms_key.gd_key"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsGuardDutyPublishingDestinationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsGuardDutyPublishingDestinationConfig_basic(bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsGuardDutyPublishingDestinationExists(resourceName),
+					resource.TestCheckResourceAttrPair(resourceName, "detector_id", detectorResourceName, "id"),
+					resource.TestCheckResourceAttrPair(resourceName, "destination_arn", bucketResourceName, "arn"),
+					resource.TestCheckResourceAttrPair(resourceName, "kms_key_arn", kmsKeyResourceName, "arn"),
+					resource.TestCheckResourceAttr(resourceName, "destination_type", "S3")),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAwsGuardDutyPublishingDestination_disappears(t *testing.T) {
+	resourceName := "aws_guardduty_publishing_destination.test"
+	bucketName := acctest.RandomWithPrefix("tf-acc-test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsGuardDutyPublishingDestinationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsGuardDutyPublishingDestinationConfig_basic(bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsGuardDutyPublishingDestinationExists(resourceName),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsGuardDutyPublishingDestination(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccAwsGuardDutyPublishingDestinationConfig_basic(bucketName string) string {
+	return fmt.Sprintf(`
+
+data "aws_caller_identity" "current" {}
+
+data "aws_region" "current" {}
+
+data "aws_iam_policy_document" "bucket_pol" {
+  statement {
+    sid = "Allow PutObject"
+    actions = [
+      "s3:PutObject"
+    ]
+
+    resources = [
+      "${aws_s3_bucket.gd_bucket.arn}/*"
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["guardduty.amazonaws.com"]
+    }
+  }
+
+  statement {
+    sid = "Allow GetBucketLocation"
+    actions = [
+      "s3:GetBucketLocation"                                                   
+    ]
+
+    resources = [
+      "${aws_s3_bucket.gd_bucket.arn}"
```

Comment:
As the bot comment above mentions, Terraform 0.12 syntax in the testing and documentation is now required. 👍 

e.g.

```suggestion
      aws_s3_bucket.gd_bucket.arn
```